src/runtime_src/tools/xclbinutil/SectionIPLayout.cxx

```diff
@@ -119,6 +119,75 @@ SectionIPLayout::getIPControlType(std::string& _sIPControlType) const
   throw std::runtime_error(errMsg);
 }
 
+const std::string
+SectionIPLayout::getFunctionalStr(PS_FUNCTIONAL eFunctional) const
+{
+  switch (eFunctional) {
+    case FC_DPU:
+      return "DPU";
+    case FC_PREPOST:
+      return "PrePost";
+    default:
+      return (boost::format("UNKNOWN (%d)") % static_cast<unsigned int>(eFunctional)).str();
+  }
+}
+
+PS_FUNCTIONAL
+SectionIPLayout::getFunctionalNoError(std::string& sFunctional)
+{
+  if (sFunctional == "DPU") 
+    return FC_DPU;
+  else if (sFunctional == "PrePost") 
+    return FC_PREPOST;
+  else
+    return FC_UNKNOWN;
+}
+
+PS_FUNCTIONAL
+SectionIPLayout::getFunctional(std::string& sFunctional) const
+{
+  PS_FUNCTIONAL eFunctional = getFunctionalNoError(sFunctional); 
```

Comment:
Please refactor this call to be in a try/catch block.  If the string value is invalided, then an exception should be thrown and caught.

Reply:
code refactored